### DISCUSSION TITLE
Night Light Mode idea + mockup for Floorp

### DISCUSSION THREAD

**Discussion by gunir** [Floorp-Projects/Floorp#481] [Polls]
Because currently we doesn't have a built-in Dark Mode but many of us want Dark Mode, so at least a Night Light mode can reduce eye strain by a decent margin.

By researching Android and Flux's Night Light mode, I understand that transparent amber background can reduce eye strain, which help browsing, [reduce eye strain in low-light conditions](http://people.ds.cam.ac.uk/ssb22/css/dark.html), and also improve sleep condition.

Solution: A button to toggle this CSS, it's surprisingly easy but incredibly helpful:
- On: `#main-window:hover {opacity: 0.7!important;background: #ffbf00!important;}`
- Off: `#main-window:hover {opacity: unset!important;background: unset!important;}`

You can use Userscript to modify `#main-window` too, I recommend this solution because we need to check if Floorp's `#main-window` is loaded or not, if it's loaded then change css and userChrome method can't because modifing `#main-window` too early may cause browser to get completely transparent and click through-able:
On:
```
document.getElementById('main-window').style.opacity = '0.7'
document.getElementById('main-window').style.background = '#ffbf00'
```

Off:
```
document.getElementById('main-window').style.opacity = 'unset'
document.getElementById('main-window').style.background = 'unset'
```

How does it look ?
![image](https://github.com/Floorp-Projects/Floorp/assets/134402102/b04d9985-1f05-4f89-b10c-0ff845001bc2)

Thanks for making Floorp, hopefully this will be helpful, I love Floorp and always want to contribute something to help Floorp!

**Comment by creeper-0910** [comment]
Another option is to use the standard Windows night mode.
